### DISCUSSION TITLE
--set-default-version is an invalid command

### DISCUSSION THREAD

**Discussion by mforkel** [microsoft/WSL#5790] [General]
# Environment

```
Windows 10 Pro, Version 1909, Build 10.0.18363.1016
```

# Steps to reproduce

```
PS C:\> wsl --set-default-version 2
Ungültige Befehlszeilenoption: --set-default-version
```

#  Expected behavior

Switch from WSL 1 to WSL 2

# Actual behavior

Error message, unknow option --set-default-version

According to [WSL 2 Support is coming to Windows 10 Versions 1903 and 1909](https://devblogs.microsoft.com/commandline/wsl-2-support-is-coming-to-windows-10-versions-1903-and-1909/), WSL 2 should be available for build 18362. I followed the update instructions in  [Windows Subsystem for Linux Installation Guide for Windows 10](https://docs.microsoft.com/en-us/windows/wsl/install-win10), including the installation of Windows Subsystem for Linux Update - 10.19.104. But I get an 'Invalid command line option' error when it run  `wsl --set-default-version 2`.

**Comment by benhillis** [comment]
Are you sure you have all Windows updates? You will need to reboot after installing them.

**Comment by mforkel** [comment]
Yes, I'm sure. The last update, installed at the same time as Windows Subsystem for Linux Update - 10.19.104, was [KB4570723](https://support.microsoft.com/en-us/help/4570723/kb4570723-cumulative-update-for-net-framework). I've been running Windows Update often lately, hoping that the update to 2004 would become available for my system ... :-)

**Comment by vuppalaprasanth** [comment]
Its only after OS build 18362.1049. **Minor Build number greater than 1049.**  
Hope you had that release installed.   
If you have that build installed then rebooting the system would work.

**Comment by mforkel** [comment]
Ok, my minor build number is too low. My fault. I didn't read the instructions closely enough.

How do I get from 18363.1016 to 18363.1049 if Windows Update does not find any updates? Can I / should I install [KB4566116](https://support.microsoft.com/en-us/help/4566116/windows-10-update-kb4566116) by hand to get to 1049?

**Comment by WSLUser** [comment]
So looks like there is an issue with Windows Update not bumping the build number or not applying the KB correctly. It literally was the fastest update I had done and it showed the Linux kernel was installed (though it needs to be bumped to the latest in the 4.19x series or better yet 5.4 series). Still ran into this issue however (I happened by coincidence to revert to previous build after the 2004 August patches broke the ability to boot Windows).

**Comment by Ivoz** [comment]
After you get to the 20.04 Windows version then WSL2 should run okay.

If you click Check For Updates regularly I think that might tell your Windows try to ask for 2004 build sooner.

**Comment by mforkel** [comment]
After some reading, my understanding is this: While the backport has been [announced](https://devblogs.microsoft.com/commandline/wsl-2-support-is-coming-to-windows-10-versions-1903-and-1909/), the required update [KB4566116](https://support.microsoft.com/en-us/help/4566116/windows-10-update-kb4566116) was released as a preview. I assume it will become availabe as a regular update on September patch day.

**Comment by licanhua** [comment]
I didn't do the testing, but it's possible for you to install it http://www.catalog.update.microsoft.com/Search.aspx?q=KB4566116 manually.

If you have WSL 1 distros. I would suggest you backup these reg keys before the installation. It will allow you to recover the distros if distros are missing after you apply the update.
 
```
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Lxss
HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\Lxss
```

**Comment by mwpowellhtx** [comment]
I seem to be stuck in Version 1909 OS Build 18363.1016 Neverneverland vis-a -vis my ability to obtain WSL 2 support. I have attempted to install the [KB4566116](http://www.catalog.update.microsoft.com/Search.aspx?q=KB4566116) and install manually, to no avail. Any ideas? I think I am "up to date" on Windows updates, otherwise. Thank you...

**Comment by ykfq** [comment]
Not working for `1909 18363.592`:
![image](https://user-images.githubusercontent.com/20741596/100173412-70dff700-2f05-11eb-9db0-23b090f7c7a1.png)

Still need to install the KB4566116 manually

**Comment by mwpowellhtx** [comment]
@ykfq As far as I know, yes, if you can.  I could not find a way to do it; would not install for me, and I am fairly certain I isolated the _KB_ download. Anyway, I found a way to update the _OS_ itself to `2004 OS Build 19041.572`, which, as far as I know, allowed me to continue. No idea what the contingencies are fo _i8n_ versions.
